### PR TITLE
refactor: Add add_tags() and get_tags() to ProblemReport

### PR DESCRIPTION
**Pull request by bdrung**
There are multiple Apport hooks that want to add tags to the report. Add `add_tags` and `get_tags` to `ProblemReport` to avoid code duplication.

### CODE REVIEW

File: apport/ui.py [hunk 1]
```diff
@@ -1982,10 +1978,7 @@ def add_extra_tags(self):
         """Add extra tags to report specified with --tags on CLI."""
         assert self.report
         if self.args.tag:
-            tags = self.report.get("Tags", "")
-            if tags:
-                tags += " "
-            self.report["Tags"] = tags + " ".join(self.args.tag)
+            self.report.add_tags(self.args.tag)
```
Comment (schopin-pro) [inline apport/ui.py:1981]: question: the variable is `tag`, singular. Shouldn't it be put in an array instead?

**Review by schopin-pro — COMMENTED** [review]
Just a small question, otherwise LGTM.

File: apport/ui.py (hunk 1 above)
Comment (bdrung) [reply apport/ui.py:1981]: `args.tag` is a list:
```python
        parser.add_argument(
            "--tag",
            action="append",
            default=[],
```
Should I rename `args.tag` to `args.tags` to reflecting that?

Comment (schopin-pro) [reply apport/ui.py:1981]: If that's easily feasible, then yeah I think it'd make sense, but don't got too much out of your way.

Comment (bdrung) [reply apport/ui.py:1981]: Let me do it. It should be easy to change.

Comment (bdrung) [reply apport/ui.py:1981]: Ready: https://github.com/canonical/apport/pull/73

I will rebase this merge request once PR #73 is reviewed and merged.

**Pull request merged**